### PR TITLE
DM-29299: Clean up physicalFilterMap and fix background retrieval

### PR DESCRIPTION
**Pull request by erykoff**
This fixes two bugs.  The first is that the fgcm filterToBand expects no extra filters, so the full physicalFilterMap needs to be pruned.  The second is the background detector retrieval in gen3.

### CODE REVIEW

File: python/lsst/fgcmcal/fgcmBuildStarsBase.py [hunk 1]
```diff
@@ -607,9 +609,9 @@ def _fillVisitCatalog(self, visitCat, groupedDataRefs, bkgDataRefDict=None,
                         bgList = dataRef.get(datasetType='calexpBackground')
                         foundBkg = True
                 else:
-                    det = dataRef.dataId['detector']
                     try:
-                        bkgRef = bkgDataRefDict[(visit, det)]
+                        # Use the same detector used from the summary.
```
Comment (laurenam) [inline python/lsst/fgcmcal/fgcmBuildStarsBase.py:613]: A "Gen3..." comment as on L559 above for this `else` block could be helpful.

File: python/lsst/fgcmcal/fgcmBuildStarsBase.py [hunk 2]
```diff
@@ -564,6 +564,8 @@ def _fillVisitCatalog(self, visitCat, groupedDataRefs, bkgDataRefDict=None,
                     # Take the first available ccd if reference isn't available
                     summaryRow = summary[0]
 
+                summaryDetector = summaryRow['id']
```
Comment (laurenam) [inline python/lsst/fgcmcal/fgcmBuildStarsBase.py:567]: Remove space above (just 'cuz more spaces mean longer file!)

File: python/lsst/fgcmcal/fgcmBuildStarsBase.py [hunk 3]
```diff
@@ -516,15 +516,15 @@ def _fillVisitCatalog(self, visitCat, groupedDataRefs, bkgDataRefDict=None,
...
-           Dataref to write visitCat for checkpoints
-        bkgDataRefDict: `dict`, optional
-           Dictionary of gen3 dataRefHandles for background info. FIXME
+        visitCat : `afw.table.BaseCatalog`
+            Catalog with schema from _makeFgcmVisitSchema()
+        groupedDataRefs : `dict`
+            Dictionary with visit keys, and `list`s of
+            `lsst.daf.persistence.ButlerDataRef`
+        visitCatDataRef : `lsst.daf.persistence.ButlerDataRef`, optional
+            Dataref to write visitCat for checkpoints
```
Comment (laurenam) [inline python/lsst/fgcmcal/fgcmBuildStarsBase.py:525]: Double backticks around `visitCat` and end with a period.

File: python/lsst/fgcmcal/fgcmBuildStarsBase.py [hunk 4]
```diff
@@ -516,15 +516,15 @@ def _fillVisitCatalog(self, visitCat, groupedDataRefs, bkgDataRefDict=None,
...
-        visitCatDataRef: `lsst.daf.persistence.ButlerDataRef`, optional
-           Dataref to write visitCat for checkpoints
-        bkgDataRefDict: `dict`, optional
-           Dictionary of gen3 dataRefHandles for background info. FIXME
+        visitCat : `afw.table.BaseCatalog`
+            Catalog with schema from _makeFgcmVisitSchema()
+        groupedDataRefs : `dict`
+            Dictionary with visit keys, and `list`s of
+            `lsst.daf.persistence.ButlerDataRef`
+        visitCatDataRef : `lsst.daf.persistence.ButlerDataRef`, optional
```
Comment (laurenam) [inline python/lsst/fgcmcal/fgcmBuildStarsBase.py:524]: Can these also be whatever the Gen3 version of `lsst.daf.persistence.ButlerDataRef` is (and for `groupedDataRefs` above)?

File: python/lsst/fgcmcal/utilities.py [hunk 5]
```diff
@@ -64,6 +65,8 @@ def makeConfigDict(config, log, camera, maxIter,
         Reset fit parameters before fitting?
     outputZeropoints: `bool`
         Compute zeropoints for output?
+    lutFilterNames : array-like, `str`
+        Array of physical filter names in the LUT
```
Comment (laurenam) [inline python/lsst/fgcmcal/utilities.py:69]: End with period.

File: tests/config/fgcmFitCycleHsc.py [hunk 6]
```diff
@@ -1,7 +1,10 @@
 import lsst.fgcmcal as fgcmcal
 
 config.outfileBase = 'TestFgcm'
-config.physicalFilterMap = {'HSC-G': 'g', 'HSC-R': 'r', 'HSC-I': 'i'}
+config.physicalFilterMap = {'HSC-G': 'g',
+                            'HSC-R': 'r',
+                            'HSC-I': 'i',
+                            'HSC-Z': 'z'}
```
Comment (laurenam) [inline tests/config/fgcmFitCycleHsc.py:7]: Why add the `z` band mapping here if you don't use it (that I can see) in the tests?

File: python/lsst/fgcmcal/fgcmBuildStarsBase.py [hunk 7]
```diff
@@ -516,15 +516,15 @@ def _fillVisitCatalog(self, visitCat, groupedDataRefs, bkgDataRefDict=None,
...
-           Catalog with schema from _makeFgcmVisitSchema()
-        groupedDataRefs: `dict`
-           Dictionary with visit keys, and `list`s of
-           `lsst.daf.persistence.ButlerDataRef`
-        visitCatDataRef: `lsst.daf.persistence.ButlerDataRef`, optional
-           Dataref to write visitCat for checkpoints
-        bkgDataRefDict: `dict`, optional
-           Dictionary of gen3 dataRefHandles for background info. FIXME
+        visitCat : `afw.table.BaseCatalog`
+            Catalog with schema from _makeFgcmVisitSchema()
```
Comment (laurenam) [inline python/lsst/fgcmcal/fgcmBuildStarsBase.py:520]: End with period.   I'm not totally clear on this...is it just the schema that is needed, or is it that the schemas must match? (pre-existing, so feel free to ignore!)

**Review by laurenam — APPROVED** [review]

File: python/lsst/fgcmcal/fgcmBuildStarsBase.py (hunk 4 above)
Comment (erykoff) [reply python/lsst/fgcmcal/fgcmBuildStarsBase.py:524]: `groupedDataRefs` can be either (updated), but `visitCatDataRef` is Gen2 only, and a note has been added.

File: tests/config/fgcmFitCycleHsc.py (hunk 6 above)
Comment (erykoff) [reply tests/config/fgcmFitCycleHsc.py:7]: I've added a specific note that this is here to test the failure that is being fixed (namely, that if there were extra filters defined in the `physicalFilterMap` then `fgcm` would complain.)

File: python/lsst/fgcmcal/fgcmBuildStarsBase.py (hunk 7 above)
Comment (erykoff) [reply python/lsst/fgcmcal/fgcmBuildStarsBase.py:520]: I've changed this to `Visit catalog.  See _makeFgcmVisitSchema() for schema definition.`

**Pull request merged**